Updated sync documentation

simon-ball opened the pull request:
https://github.com/kavli-ntnu/dj-moser-imaging/issues/111

simon-ball committed:
Update sync page

* Move sync images to their own folder
* Expand on explanation of sync data
* Point to Setup.Sync for identifying what each series means

simon-ball committed:
Update sync.rst

simon-ball committed:
Additional information about sync_name and generic_name

horsto commented on docs/imaging/sync.rst:
```diff
@@ -4,31 +4,117 @@
+When a new Session is ingested, the synchronisation data is inserted into the ``Sync`` table and its part tables. Each Session has one (or more) entries in the ``Sync`` table, corresponding to a single time series, from a single instrument, with each time series being identified by a ``sync_name``.
+
+The actual timestamps themselves are stored in the column ``sync_data`` of the ``Sync`` table. The values are stored in *samples*. The sample rate (samples per second) is also stored, in column ``sample_rate``, if you wish to convert into time periods in seconds. 
+
```
Add: Note that all sync data that is recorded by scanimage itself is in seconds already (`sample_rate = 1`).

horsto commented on docs/imaging/sync.rst:
```diff
@@ -4,31 +4,117 @@
+
+The most important attribute here is the ``generic_name``. This defines how further downstream tables identfy *which* Sync stream they should be looking for, to pair time-series data with its timestamps. The main ``generic_names`` are as follows:
+
+* ``frames_imaging`` : timestamps for two-photon imaging frames
```
..., defined as the start time of scanning on each frame. For multi plane recordings this resolves to the start time of scanning of each volume (=stack), such that the number of data points saved here matches the number of distinct timepoints recorded for single cells (i.e. number of data points in sync equals the number of data points in imaging analysis output like suite2p for every cell).

horsto commented on docs/imaging/sync.rst:
```diff
@@ -4,31 +4,117 @@
+* ``Tracking2LED`` : timestamps of video frames of open-field tracking, where the subject is identified by a pair of head-mounted LEDs
+* ``TrackingDLC`` : timestamps of video frames intended to be analysed with DLC (see :ref:`Imaging dlc`).
+
+``sync_name`` is intended to provide a name that conveys more information to the users.
```
I would write: ... is the name that the sync data is called in the raw data files (e.g. scanimage TIF file, wavesurfer sync files, etc.). For scanimage TIFs this name is derived from the timing information that is found in the header of each TIF page (i.e. frame timestamps, AUX and I2C timestamps etc.)

horsto commented on docs/imaging/sync.rst:
```diff
@@ -4,31 +4,117 @@
+* ``tracking_rotary`` : timestamps of 1-dimensional, wheel-encoder data
+* ``Tracking2LED`` : timestamps of video frames of open-field tracking, where the subject is identified by a pair of head-mounted LEDs
+* ``TrackingDLC`` : timestamps of video frames intended to be analysed with DLC (see :ref:`Imaging dlc`).
+
```
Add: For example, the `SignalTracking` make() routine looks for the `frames_imaging` and either `Tracking2LED` or `TrackingDLC` related sync streams to synchronise imaging and tracking data, which can then be used to create spatial tuning maps / curves.

horsto commented on docs/imaging/sync.rst:
```diff
@@ -4,31 +4,117 @@
+    fig, ax = plt.subplots()
+    ax.plot(sync_data_s, df_f)
+    ax.set_ylabel("df/f")
+    ax.set_xlabel("time (s)")
```
Some time ago I wrote a helper for this https://github.com/kavli-ntnu/dj-moser-imaging/blob/de303b0bf293a59c78280547f88c4de66b749a8a/dj_schemas/utils.py#L529 

It might be good to provide this function as part of the user_imaging repo (future goal?).

horsto commented on docs/imaging/sync.rst:
```diff
@@ -4,31 +4,117 @@
\ No newline at end of file
+The OpenMiniscope system implements one particular special case of the more general synchronisation scheme outlined above. Data can be *pre-synchronised* with the `ScanImage <http://scanimage.vidriotechnologies.com/display/SIH/ScanImage+Home>`_ software. In this case, instead of each instrument operating at its own sampling frequency determined by its own independent clock, each frame of acquisit ...
+
+In this case, the timestamp data for two-photon imaging frames is identical to the timestamp data for tracking frames.
```
For example, a tracking camera. In this case, the timestamp data ... .

horsto commented on docs/imaging/sync.rst:
```diff
@@ -4,31 +4,117 @@
+In this case, the timestamp data for two-photon imaging frames is identical to the timestamp data for tracking frames.
+
+Note that although this method is superficially much simpler than the more general case, it has its own complexities, especially where multi-plane two-photon imaging is in use. In general, the design of the OpenMiniscope system is to send one trigger to external instruments on each *volume* acquisition, rather than on each *plane* acquisition, and consequently, there may be an integer multiple be ...
+
```
no, the tracking is triggered on each "plane", for example, if the volume rate is 7 Hz for 2 planes, the tracking camera runs at 14 Hz.

horsto commented on docs/imaging/sync.rst:
```diff
@@ -4,31 +4,117 @@
+Note that although this method is superficially much simpler than the more general case, it has its own complexities, especially where multi-plane two-photon imaging is in use. In general, the design of the OpenMiniscope system is to send one trigger to external instruments on each *volume* acquisition, rather than on each *plane* acquisition, and consequently, there may be an integer multiple be ...
+
+For instruments where the acquisition rate of the two-photon imaging frames is too slow, the system supports acquiring a limited number of data streams at the acquisition rate of the two-photon photodetectors. In this case, a much higher temporal resolution can be saved as an "imaging frame", generating a TIF image (essentially, an overengineered array of data) alongisde the imaging data.
+    
```
" In this case, a much higher temporal resolution can be saved as an "imaging frame", generating a TIF image (essentially, an overengineered array of data) alongisde the imaging data." -> No, this doesn't increase the temporal resolution at all (since timestamps within (!) a frame acquisition cannot be resolved).

Better to refer to the AUX and I2C signals here which generate timestamps at scanimage's internal clock speed and are therefore arbitrarily precise compared to signals locked to the imaging acquisition itself.

horsto reviewed (changes requested):
Thanks, I think this clarifies things a lot more. See my (few) comments and suggestions for improvement.

simon-ball committed:
Create extracted_sync_data.png

simon-ball committed:
Address PR comments

simon-ball committed:
Add information on high-speed-sync with ScanImage pre-sync

NB: I need to get Weijian to read over this for accuracy. This is based off my best understanding of his approach, but it's definitely 2nd hand.
